CRM/Odoosync/Sync/Setting.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * Retrieves the settings created by the extension
+ */
+class CRM_Odoosync_Sync_Setting {
+
+  /**
+   * Instance for singleton
+   *
+   * @var object
+   */
+  protected static $instance;
+
+  /**
+   * Setting for Odoo sync
+   *
+   * @var array
+   */
+  protected $setting;
+
+  /**
+   * Required setting fields for the extension
+   */
+  private $requiredSettingFields = [
+    'odoosync_odoo_instance_url',
+    'odoosync_database_name',
+    'odoosync_username',
+    'odoosync_password',
+    'odoosync_batch_size',
+    'odoosync_retry_threshold',
+    'odoosync_error_notice_address',
+  ];
+
+  /**
+   * CRM_Odoosync_Sync_Setting constructor.
+   *
+   * @throws \Exception
+   */
+  protected function __construct() {
+    $this->setSettings();
+    $this->validateSetting();
+  }
+
+  /**
+   * Gets singleton instance
+   */
+  public static function getInstance() {
```

Comment:
No, we are using *CRM_Odoosync_Setting::getInstance()* in four classes:
* CRM/Odoosync/Sync/Request/Contact.php
* CRM/Odoosync/Sync/Request/Auth.php
* CRM/Odoosync/Sync/BaseHandler.php
* Also, it will be used for contribution sync